Incomplete Bug Fix of "Unable to build application due to missing type declarations. #415"

### Echo Version

2.0.2

### Laravel Version

12.0.1

### PHP Version

8.3

### NPM Version

11.1.0

### Database Driver & Version

_No response_

### Description

The changes made in https://github.com/laravel/echo/pull/416 fixed the issue in my project but on a fresh project the error appears to still exists.


```
$ npm run build
> build
> tsc && vite build

Error: node_modules/laravel-echo/dist/channel/socketio-channel.d.ts(3,29): error TS2307: Cannot find module 'socket.io-client' or its corresponding type declarations.
Error: node_modules/laravel-echo/dist/connector/socketio-connector.d.ts(3,33): error TS2307: Cannot find module 'socket.io-client' or its corresponding type declarations.
Error: Process completed with exit code 2.
```
(https://github.com/fxnm/laravel-echo-bug-report/actions/runs/13605936877/job/38037408383)

### Steps To Reproduce

Bug Report Repo: https://github.com/fxnm/laravel-echo-bug-report
 
- Create a laravel application with echo + typescript.
- Set in the tsconfig.json the option "skipLibCheck": false
- Try to create a frontend build using npm run build and fail :(

--- a/package.json
+++ b/package.json
@@ -55,6 +55,10 @@
         "tslib": "^2.8.1",
         "typescript": "^5.7.0"
     },
+    "peerDependencies": {
+        "socket.io-client": "*",
+        "pusher-js": "*"
+    },
     "typesVersions": {
         "*": {
             "socket.io-client": [],

